cockatrice/src/tab_server.cpp

```diff
@@ -178,3 +155,47 @@ void TabServer::processServerMessageEvent(const Event_ServerMessage &event)
     serverInfoBox->setHtml(QString::fromStdString(event.message()));
     emit userEvent();
 }
+
+void TabServer::joinRoom(int id, bool setCurrent)
+{
+    TabRoom *room = tabSupervisor->getRoomTabs().value(id);
+    if(!room)
+    {
+        Command_JoinRoom cmd;
+        cmd.set_room_id(id);
+        
+        PendingCommand *pend = client->prepareSessionCommand(cmd);
+        pend->setExtraData(setCurrent);
+        connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(joinRoomFinished(Response, CommandContainer, QVariant)));
+        
+        client->sendCommand(pend);
+
+        return;   
+    }
+
+    if(setCurrent)
+        tabSupervisor->setCurrentWidget((QWidget*)room);
+}
+
+void TabServer::joinRoomFinished(const Response &r, const CommandContainer & /*commandContainer*/, const QVariant &extraData)
+{
+    switch (r.response_code()) {
+        case Response::RespOk:
+            break;
+        case Response::RespNameNotFound:
+            QMessageBox::critical(this, tr("Error"), tr("Failed to join the room: it doesn't exists on the server."));
+            return;
+        case Response::RespContextError:
+            QMessageBox::critical(this, tr("Error"), tr("You are already inside the room."));
+            return;
+        case Response::RespUserLevelTooLow:
+            QMessageBox::critical(this, tr("Error"), tr("You do not have the proper permission to join this room."));
+            return;
+        default:
+            QMessageBox::critical(this, tr("Error"), tr("Failed to join the room due to an unknown error."));
```

Comment:
Can you display the `response_code` here so we can debug it easier? Maybe also a `qDebug`